tools/rewriter/GenCallAsm.cpp

```diff
@@ -915,6 +915,24 @@ static void emit_restore_args(AsmWriter &aw, Arch arch, bool pop) {
   }
 }
 
+static void emit_align_stack(AsmWriter &aw, Arch arch) {
+  add_comment_line(aw, "Align stack");
+  if (arch == Arch::X86) {
+    add_asm_line(aw, "subq $8, %rsp");
+  } else if (arch == Arch::Aarch64) {
+    // TODO
+  }
+}
+
+static void emit_revert_align_stack(AsmWriter &aw, Arch arch) {
+  add_comment_line(aw, "Revert align stack");
+  if (arch == Arch::X86) {
+    add_asm_line(aw, "addq $8, %rsp");
+  } else if (arch == Arch::Aarch64) {
+    // TODO
+  }
+}
+
```

Comment:
These should have some documentation or clearer names to indicate that they don't *always* align the stack, they just adjust it by 8. The calling context needs to know the current stack misalignment to know whether calling `emit_align_stack` will align the stack or not.

Reply:
Fixed in ba28321.  How does that look?  I'm not really sure what a great name would be, though.